src/target/renesas.c

```diff
@@ -571,15 +589,21 @@ static bool renesas_rv40_flash_write(target_flash_s *f, target_addr_t dest, cons
 		len -= write_size;
 
 		/* issue two part Write commands */
-		target_mem_write8(t, RV40_CMD, RV40_CMD_PROGRAM);
-		target_mem_write8(t, RV40_CMD, (uint8_t)(write_size / 2U));
+		if (option_flash) {
+			target_mem_write8(t, RV40_CMD, RV40_CMD_CONFIG_SET_1);
+			target_mem_write8(t, RV40_CMD, RV40_CMD_CONFIG_SET_2);
+		} else {
+			target_mem_write8(t, RV40_CMD, RV40_CMD_PROGRAM);
+			target_mem_write8(t, RV40_CMD, (uint8_t)(write_size / 2U));
+		}
 
 		/* according to reference manual the data buffer full time for 2 bytes is 2 usec.
 		 * this is with a FCLK of 4MHz
 		 * a complete should take less than 1 msec.
 		 */
 		platform_timeout_s timeout;
-		platform_timeout_set(&timeout, 10);
+		/* 200ms is arbitrary number i made up */
+		platform_timeout_set(&timeout, option_flash ? 200: 20);
```

Comment:
surely the expected time is documented somehow, it was for most other flash related things, why 200 over 20, was 20 too short?

Reply:
20ms was way too short, I ended up on 100ms, tested 50 gave me errors, 75 was stable, but I could only test RA4M2, not sure what other series would behave like, also this timeout should be as high as we feel good with it, this is just timeout, there is `RV40_FSTATR_RDY` check afterwards anyway, so we are not making things go slow.